profetch: init at 0.1.6

q60 opened the pull request:
profetch is a system information fetcher written in GNU/Prolog
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

q60 committed:
profetch: init at v0.1.6

q60 committed:
disfetch: more convenient way of using name/rev

OPNA2608 commented on pkgs/tools/misc/profetch/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+stdenv.mkDerivation rec {
+  pname = "profetch";
+  version = "v0.1.6";
```
```suggestion
  version = "0.1.6";
```

OPNA2608 commented on pkgs/tools/misc/profetch/default.nix:
```diff
@@ -0,0 +1,34 @@
+  src = fetchFromGitHub {
+    owner = "RustemB";
+    repo = pname;
+    rev = version;
```
```suggestion
    rev = "v${version};
```

OPNA2608 commented on pkgs/tools/misc/profetch/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+  buildInputs = [ gprolog ];
+
+  buildPhase = ''
```
```suggestion
  buildPhase = ''
    runHook preBuild
```

OPNA2608 commented on pkgs/tools/misc/profetch/default.nix:
```diff
@@ -0,0 +1,34 @@
+    gplc profetch.pl --no-top-level            \
+                     --no-debugger --no-fd-lib \
+                     --no-fd-lib-warn --min-size -o profetch
+  '';
+
+  installPhase = ''
```
```suggestion
    runHook postBuild
  '';

  installPhase = ''
    runHook preInstall
```

OPNA2608 commented on pkgs/tools/misc/profetch/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+  installPhase = ''
+    install -Dm755 -t $out/bin profetch
+  '';
```
```suggestion
    runHook postInstall
  '';
```

OPNA2608 commented on pkgs/tools/misc/profetch/default.nix:
```diff
@@ -0,0 +1,34 @@
+  meta = with lib; {
+    description = "System Information Fetcher Written in GNU/Prolog";
+    homepage = "https://github.com/RustemB/profetch";
+    license = licenses.gpl3;
```
GPL3-only or GPL3+? Please clarify.

q60 replied:
GPL3-only.

q60 committed:
profetch: update version to suit recomendations

Co-authored-by: Christoph Neidahl <christoph.neidahl@gmail.com>

q60 committed:
profetch: update version to suit recomendations

Co-authored-by: Christoph Neidahl <christoph.neidahl@gmail.com>

q60 committed:
profetch: update code alignment

Co-authored-by: Christoph Neidahl <christoph.neidahl@gmail.com>

q60 committed:
profetch: refined code

+ Added missing `"` to the `rev` variable
+ We use 4-space indentation in bash scripts inside nix code
+ License in GPL3-only
+ One-line args because not too much of them

r-rmcgibbo commented:
Result of `nixpkgs-review pr 133616` at 90cc1994 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>profetch</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild`.

Near pkgs/tools/misc/profetch/default.nix:16:3:
```
   |
16 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook postInstall`.

Near pkgs/tools/misc/profetch/default.nix:23:3:
```
   |
23 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/tools/misc/profetch/default.nix:
```diff
@@ -0,0 +1,35 @@
+    sha256 = "1clh3l50wz6mlrw9kx0wh2bbhnz6bsksyh4ngz7givv4y3g9m702";
+  };
+
+  buildInputs = [ gprolog ];
```
```suggestion
  nativeBuildInputs = [ gprolog ];
```

q60 committed:
profetch: add postInstall hook

Co-authored-by: Christoph Neidahl <christoph.neidahl@gmail.com>

q60 committed:
profetch: add preBuild hook

Co-authored-by: Christoph Neidahl <christoph.neidahl@gmail.com>

q60 committed:
profetch: replace buildInputs with nativeBuildInputs

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

q60 committed:
profetch: indentation fix

SuperSandro2000 commented on pkgs/tools/misc/profetch/default.nix:
```diff
@@ -23,6 +23,7 @@ stdenv.mkDerivation rec {
   installPhase = ''
       runHook preInstall
       install -Dm755 -t $out/bin profetch
```
```suggestion
    runHook preInstall
    install -Dm755 -t $out/bin profetch
```

q60 committed:
Update default.nix

q60 commented:
anybody?

figsoda commented:
Result of `nixpkgs-review pr 133616` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>profetch</li>
  </ul>
</details>
